frontend/src/metabase/parameters/components/widgets/ParameterFieldWidget/FieldValuesWidget/ListField/utils.ts

```diff
@@ -1,3 +1,31 @@
-export function isValidOptionItem(optionItem: any, filter: string): boolean {
-  return String(optionItem).toLowerCase().includes(filter);
+import type { ContentTranslationFunction } from "metabase/i18n/types";
+import type { FieldValue } from "metabase-types/api";
+
+/**
+ * Returns true when the option matches the filter. If the option has an
+ * available content translation, the filter is tested against this translation
+ **/
+export function optionMatchesFilter(
+  option: FieldValue | undefined,
+  filter: string,
+  translate?: ContentTranslationFunction,
+) {
+  // If filter has no diacritics (like äéĩøů), let's ignore diacritics in the
+  // option
+  const ignoreDiacritics = filter === removeDiacritics(filter);
+
+  return option?.some((value) => {
+    const stringifiedValue = String(value ?? "");
+    const maybeTranslatedValue =
+      translate?.(stringifiedValue) || stringifiedValue;
+    const normalizedValue = ignoreDiacritics
+      ? removeDiacritics(maybeTranslatedValue)
+      : maybeTranslatedValue;
+    return normalizedValue.toLowerCase().includes(filter);
+  });
 }
```

Comment:
this is very fancy, but I'm wondering what performance implications it has on large field lists